docker/fetch_normalization_data.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+
+"""
+Fetches normalization data.
+
+Usage:
+
+    python docker/fetch_normalization_data.py
+
+"""
+
+import argparse
+import datetime
+import os
+import sys
+
+import psycopg2
+import requests
+
+
+# NOTE(willkg): These platforms come from the -prod data.
+PLATFORMS = ['Mac OS X', 'Linux', 'Windows', 'Unknown']
+
+
+def fetch_data_from_api(endpoint, product, version, platform, start_date, end_date):
+    """Fetches data from API"""
+    req = requests.get(endpoint, params={
+        'end_date': end_date,
+        'start_date': start_date,
+        'product': product,
+        'platforms': platform,
+        'versions': version,
+    })
+    if req.status_code != 200:
+        print('Error retrieving ADI: %s' % req.content)
+        return []
+
+    data = req.json()
+    return data['hits']
+
+
+def fetch_versions(conn, product_name):
+    """Fetch the version strings for a product from product_versions table"""
+
+    cursor = conn.cursor()
+    cursor.execute("""
+        SELECT version_string
+        FROM product_versions
+        WHERE
+            product_name=%s
+    """, (product_name,))
+    # NOTE(willkg): We don't want versions that end with "b" because they're weird.
+    versions = set([record[0] for record in cursor if not record[0].endswith('b')])
+    return list(versions)
+
+
+def fetch_product_version_id(conn, product_name, version_string):
+    """Given a product name and a version_string, returns a product_version_id"""
+    cursor = conn.cursor()
+    cursor.execute("""
+        SELECT product_version_id
+        FROM product_versions
+        WHERE
+            product_name=%s AND
+            version_string=%s
+    """, (product_name, version_string))
+    ret = cursor.fetchone()
+    if ret is None:
+        return None
+    return ret[0]
+
+
+def insert_adu(conn, adu_count, adu_date, product_name, os_name, version_string):
+    """Inserts ADU data into product_adu table"""
+    # Fetch the product_version_id we need
+    product_version_id = fetch_product_version_id(conn, product_name, version_string)
+
+    if product_version_id is None:
+        print('No product_version_id for %s %s' % (product_name, version_string))
+        return
+
+    # Insert the data
+    cursor = conn.cursor()
+    try:
+        cursor.execute("""
+            INSERT INTO product_adu (adu_count, adu_date, os_name, product_version_id)
```

Comment:
The `raw_adi` table requires data that's not available via the API endpoint, so it's not possible to do that.